processor/src/decorators/mod.rs

```diff
@@ -350,6 +421,119 @@ mod tests {
+        let empty = EmptySubtreeRoots::empty_hashes(64);
+        let initial_root = Word::from(empty[0]);
+        let seed = Blake3_256::hash(b"some-seed");
+
+        // compute pseudo-random key/value pair
+        let key = <[u8; 8]>::try_from(&(*seed)[..8]).unwrap();
+        let key = u64::from_le_bytes(key);
+        let key = [Felt::new(key); WORD_SIZE];
+        let value = <[u8; 8]>::try_from(&(*seed)[8..16]).unwrap();
+        let value = u64::from_le_bytes(value);
+        let value = [Felt::new(value); WORD_SIZE];
+
+        fn assert_case(
+            key: Word,
+            value: Word,
+            node: Word,
+            root: Word,
+            store: MerkleStore,
+            expected_stack: &[Felt],
+        ) {
+            // build the process
+            let stack_inputs = StackInputs::try_from_values([
+                root[0].as_int(),
+                root[1].as_int(),
+                root[2].as_int(),
+                root[3].as_int(),
+                key[0].as_int(),
+                key[1].as_int(),
+                key[2].as_int(),
+                key[3].as_int(),
+            ])
+            .unwrap();
+            let advice_inputs = AdviceInputs::default()
+                .with_merkle_store(store)
+                .with_map([(node.into_bytes(), value.into_iter().collect())]);
+            let advice_provider = MemAdviceProvider::from(advice_inputs);
+            let mut process = Process::new(Kernel::default(), stack_inputs, advice_provider);
+
+            // call the injector and clear the stack
+            process.execute_op(Operation::Noop).unwrap();
+            process.execute_decorator(&Decorator::Advice(AdviceInjector::SmtGet)).unwrap();
+            for _ in 0..8 {
+                process.execute_op(Operation::Drop).unwrap();
+            }
+
+            // expect the stack output
+            for _ in 0..expected_stack.len() {
+                process.execute_op(Operation::AdvPop).unwrap();
+            }
+            assert_eq!(build_expected(expected_stack), process.stack.trace_state());
+        }
+
+        // check leaves on empty trees
+        for depth in [16, 32, 48, 64] {
+            // compute the remaining key
+            let mut remaining = key;
+            remaining[3] = Felt::new(remaining[3].as_int() >> depth.min(63));
+
+            // compute node value
+            let depth = Felt::new(depth as u64);
+            let store = MerkleStore::new();
+            let node = Rpo256::merge_in_domain(&[remaining.into(), value.into()], depth).into();
+
+            // expect absent value
+            let expected = [ZERO, Felt::new(16)];
+            assert_case(key, value, node, initial_root, store, &expected);
+        }
+
+        // check leaves inserted on all tiers
+        for depth in [16, 32, 48, 64] {
+            // compute the remaining key
+            let mut remaining = key;
+            remaining[3] = Felt::new(remaining[3].as_int() >> depth.min(63));
```

Comment:
Shouldn't remaining portion of the key be computed differently? i.e., we want to take the `64 - depth` least significant bits of the element. We could do it like:
```Rust
remaining[3] = Felt::new((remaining[3].as_int() << depth) >> depth);
```
Or maybe like:
```Rust
remaining[3] = Felt::new((remaining[3].as_int() & (2.pow(depth as u32) - 1));
```

Reply:
True! Fixed to `(k3 << d) >> d`